src/gui/client.go

```diff
@@ -410,7 +410,8 @@ func (c *Client) Spend(id, dst string, coins uint64) (*SpendResult, error) {
 	v.Add("coins", fmt.Sprint(coins))
 
 	var r SpendResult
-	if err := c.Post("/wallet/spend", strings.NewReader(v.Encode()), &r); err != nil {
+	endpoint := "/wallet/spend?" + v.Encode()
```

Comment:
Why is it in the query string for a POST request?

Reply:
The spend handler was not a POST request.

Reply:
https://github.com/skycoin/skycoin/blob/develop/src/gui/wallet.go#L72-L77 ?

Reply:
Yes, the POST check was added later, but didn't change the parameters. Not sure if changing the parameter would break the exchange platform.

Reply:
It uses r.FormValue which will prefer the PostForm data but fallback on the query string.  So the gui.Client should work with POST form data

Reply:
Yes, change request's Content-Type: application/x-www-form-urlencoded, and run r.ParseForm() in the handler works.